packages/components/src/box-control/index.js

```diff
@@ -0,0 +1,134 @@
+/**
+ * External dependencies
+ */
+import { noop } from 'lodash';
+/**
+ * WordPress dependencies
+ */
+import { useInstanceId } from '@wordpress/compose';
+import { useState, useRef } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+/**
+ * Internal dependencies
+ */
+import Button from '../button';
+import { Flex } from '../flex';
+import AllInputControl from './all-input-control';
+import InputControls from './input-controls';
+import BoxControlIcon from './icon';
+import Text from '../text';
+import LinkedButton from './linked-button';
+import Visualizer from './visualizer';
+import {
+	Root,
+	Header,
+	HeaderControlWrapper,
+} from './styles/box-control-styles';
+import { useBoxControlState, isValuesMixed } from './utils';
+
+const defaultInputProps = {
+	min: 0,
+};
+
+function useUniqueId( idProp ) {
+	const instanceId = useInstanceId( BoxControl );
+	const id = `inspector-box-control-${ instanceId }`;
+
+	return idProp || id;
+}
+export default function BoxControl( {
+	id: idProp,
+	inputProps = defaultInputProps,
+	onChange = noop,
+	label = __( 'Box Control' ),
+	values: valuesProp,
+	units,
+} ) {
+	const [ values, setValues ] = useBoxControlState( valuesProp );
+
+	const [ side, setSide ] = useState( isLinked ? 'all' : 'top' );
```

Comment:
This depends on a yet-undeclared variable, `isLinked` — I don't believe it must be working. :)

The other important thing is that this is state derived from state. Is it meant to diverge from `isLinked` throughout the component's life? If so, I'd ask myself why it depends on `isLinked` in the first place and not a constant or a prop (like `! isValuedMixed( values )`). If not, I'd rewrite as a plain constant.

```js
// Option 1
const [ side, setSide ] = useState( isValuesMixed( values ) ? 'top' : 'all' );
// Option 2
const side = isLinked ? 'all' : 'top';
```

Reply:
Thank you for looking in this!

To clarify, there's also this bit:

```js
const handleOnFocus = ( event, { side: nextSide } ) => {
	setSide( nextSide );
};
```

The value of side isn't always `top` or `all`. When viewing the 4 inputs, focusing on them will change the side to `top`, `right`, `bottom`, or `left`.

In that case, is the code logic okay?

Reply:
> The value of side isn't always `top` or `all`. When viewing the 4 inputs, focusing on them will change the side to `top`, `right`, `bottom`, or `left`.
> 
> In that case, is the code logic okay?

Yeah, that makes sense. I would avoid leaving the initial state based just on `isLinked` because of what I mentioned for _option 1_ in my other comment, but that's up to you. I do recommend leaving a comment though — something to the effect of "`side`'s initial value is seeded by __ but is meant to diverge based on user input", where __ reflects the choice to keep the mention of `isLinked` or `isValuesMixed( values )`.